### ISSUE TITLE
Generics: Is everything a String?

### ISSUE THREAD

**Issue by oth-aw** [dart-lang/sdk#34634]
I'm playing with the generics. I use Dart 2.0 on MacOSX. In the code below the type "String" is accepted as type parameter. "Object" is not accepted on the other hand. Is string from every type?   

```dart
abstract class Bottle {
	int get volume;
}
class CokeBottle implements Bottle {
	 int get volume => 75;
}
class BeerBottle implements Bottle {
	 int get volume => 100;
}
class Crate<T extends Bottle> {
	List<T> bottles;
	Crate(this.bottles);
}
void main() {
 	var bottles = <BeerBottle>[BeerBottle(), BeerBottle(), BeerBottle()];
 	var crate = Crate<BeerBottle>(bottles);
 	var str = Crate<String>(['a','b','c']);
 	print(str.bottles);
}
```
Prints out
```console
[a, b, c]
```

**Comment by lrhn** [comment]
That is weird, and likely a bug.
Dart currently has two front-ends implementing the language - the analyzer and the common front-end for the compilers. Some errors have not yet been implemented in both front-ends. I believe checking of type parameter bounds is one of those. The analyzer will detect the error, but the front-end doesn't yet.

(I'm sure the front-end people can point to an existing open bug for this.)

**Comment by stefantsov** [comment]
Fixed as of aa6fb2d3a4298e60b5635f26507d5b39f4e531cb.
